pkgs/native_assets_cli/lib/src/config.dart

```diff
@@ -89,6 +89,14 @@ sealed class HookInput {
   String toString() => const JsonEncoder.withIndent('  ').convert(json);
 
   HookConfig get config => HookConfig._(this);
+
+  /// The user-defines for this [packageName].
+  HookInputUserDefines get userDefines => HookInputUserDefines._(this);
+}
+
+extension type HookInputUserDefines._(HookInput _input) {
+  /// The value for the user-define for [key] for this package.
+  Object? operator [](String key) => _input._syntax.userDefines?.json[key];
```

Comment:
It isn't exactly clear from this what types of data are allowed to be in the `userDefines`, see the other comment. Ideally this would be typed some way, if need arises we can still open this up to other data types later.

Reply:
They can be arbitrary JSON (or YAML). So List/String/Map/int/double/null. I'm not sure what is a nice way to type that.